features/step_definitions/journey_test_steps.rb

```diff
@@ -851,15 +851,19 @@ def login_page_type(user_type)
     fail("Invalid user on admin suppliers page #{@user_type}")
   end
 
-  table_rows[0].all(:css, "td").first.text.should == "DM Functional Test Supplier"
+  # Get the parent row element for the first <td> cell we found earlier
+  table_row_0 = table_row_first_cells[0].find(:xpath, 'ancestor-or-self::*[tr][1]')
+  table_row_0.all(:css, "td").first.text.should == "DM Functional Test Supplier"
   expected_links.each do |expected_link|
-    table_rows[0].all(:css, "a").map(&:text).should include(expected_link)
+    table_row_0.all(:css, "a").map(&:text).should include(expected_link)
   end
 
-  table_rows[1].all(:css, "td").first.text.should == "DM Functional Test Supplier 2"
-  expected_links.each do |expected_link|
-    table_rows[1].all(:css, "a").map(&:text).should include(expected_link)
-  end
+  # Should also be able to say the same about the following row too, but damned if I can get xpath to select it
```

Comment:
Is this something we think is important? I can have a crack at it if you think it's a useful check since I'm totally immersed in functional tests at the moment.